### PR TITLE
executor: fix batch point get return wrong result for in(null)

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/18843

Problem Summary:

### What is changed and how it works?

What's Changed:

When constucting point get index key, the datum is convert to the column type.
If the datum is null, convert it to string would construct the wrong index key.

How it Works:

Do not convert the datum if it's null.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

-   Fix bug batchPointGet returns wrong result with condition `in (null)` in some cases

### CODE REVIEW

**Comment by qw4990** [comment]
PTAL @zz-jason

**Comment by tiancaiamao** [comment]
4.0 PR https://github.com/pingcap/tidb/pull/18848

File: executor/point_get.go [hunk 1]
```diff
@@ -339,6 +339,10 @@ func (e *PointGetExecutor) get(ctx context.Context, key kv.Key) ([]byte, error)
 func EncodeUniqueIndexKey(ctx sessionctx.Context, tblInfo *model.TableInfo, idxInfo *model.IndexInfo, idxVals []types.Datum, tID int64) (_ []byte, err error) {
 	sc := ctx.GetSessionVars().StmtCtx
 	for i := range idxVals {
+		if idxVals[i].IsNull() {
```
Comment (coocood) [inline executor/point_get.go:342]: I think is no the right place to fix the bug.
The idxVals should be skipped in higher level.

Comment (coocood) [reply executor/point_get.go:342]: We should let the caller know the built key is not valid.

Comment (tiancaiamao) [reply executor/point_get.go]: Done

**Comment by coocood** [comment]
LGTM

**Comment by AndreMouche** [comment]
LGTM

**Pull request merged**